libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -852,7 +852,7 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
      * @remarks
      * Override this in a derived class to create a mock connector client for unit testing.
      */
-    protected createConnectorClient(serviceUrl: string): ConnectorClient {
+    public createConnectorClient(serviceUrl: string): ConnectorClient {
```

Comment:
I switched this to public because the `BotFrameworkAdapter.credentials` is protected, and `ConnectorClient.credentials` is not.

I also wanted the new ConnectorClient to include the `USER_AGENT`, not sure if this is a problem in C#...

Reply:
could you not make the Credentials on ConnectorClient public - that would also bring the two implementations closer together.

Reply:
The credentials in the connectorClient are [public](https://github.com/microsoft/botbuilder-js/pull/1253/files#diff-8703bbaca7e46c1a78c0da79519b20d9R79), I use them in `TeamsInfo.getTeamsConnectorClient()`

```ts
return new TeamsConnectorClient(connectorClient.credentials, { baseUri: context.activity.serviceUrl });
```

Reply:
Regarding USER_AGENT header - the C# uses the same HttpClient as the regular connector. So that should be good.
Its really just a couple of extra REST calls on top of the current connector